pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1549,6 +1549,19 @@ func parseOverlayNetworkConfig(appInstance *types.AppInstanceConfig,
 	}
 }
 
+func parseU32ConfigItem(newGlobalConfigPtr *types.GlobalConfig, key string,
+	value string, varPtr *uint32) string {
+	i64, err := strconv.ParseInt(value, 10, 32)
+	if err != nil {
+		errStr := fmt.Sprintf("bad int value %s for %s: %s\n",
+			value, key, err)
+		log.Errorf("parseU32ConfigItem: %s", errStr)
+		return errStr
+	}
+	*varPtr = uint32(i64)
```

Comment:
Please add a check if it exceeds the range for uint32 i.e., i64 >= 0 && i64 < 2^^32, and return an error in that case

Reply:
Changed it to strconv.ParseUint(value, 10, 32). This throws a "out of range" error if value exceeds 32 bits..

Reply:
You should really return int, error to fit the golang way

Reply:
But it is storing the value in *varPtr already. Why return it in success? 

This whole part with go way with the cleanup of GlobalConfig anyways..